third-party/realdds/src/dds-stream-profile.cpp

```diff
@@ -84,65 +76,63 @@ enum rs2_format  // copy from rs2_sensor.h
 
 int dds_stream_format::to_rs2() const
 {
-    static std::map< uint32_t, int > fcc_to_rs2{  // copy from ds5-device.cpp
-        { rs_fourcc( 'Y', 'U', 'Y', '2' ), RS2_FORMAT_YUYV },
-        { rs_fourcc( 'Y', 'U', 'Y', 'V' ), RS2_FORMAT_YUYV },
-        { rs_fourcc( 'U', 'Y', 'V', 'Y' ), RS2_FORMAT_UYVY },
-        { rs_fourcc( 'G', 'R', 'E', 'Y' ), RS2_FORMAT_Y8 },
-        { rs_fourcc( 'Y', '8', 'I', ' ' ), RS2_FORMAT_Y8I },
-        { rs_fourcc( 'W', '1', '0', ' ' ), RS2_FORMAT_W10 },
-        { rs_fourcc( 'Y', '1', '6', ' ' ), RS2_FORMAT_Y16 },
-        { rs_fourcc( 'Y', '1', '2', 'I' ), RS2_FORMAT_Y12I },
-        { rs_fourcc( 'Y', '1', '6', 'I' ), RS2_FORMAT_Y16I },
-        { rs_fourcc( 'Z', '1', '6', ' ' ), RS2_FORMAT_Z16 },
-        { rs_fourcc( 'Z', '1', '6', 'H' ), RS2_FORMAT_Z16H },
-        { rs_fourcc( 'R', 'G', 'B', '8' ), RS2_FORMAT_RGB8 },
-        { rs_fourcc( 'R', 'G', 'B', 'A' ), RS2_FORMAT_RGBA8 },
-        { rs_fourcc( 'R', 'G', 'B', '2' ), RS2_FORMAT_BGR8 },
-        { rs_fourcc( 'B', 'G', 'R', 'A' ), RS2_FORMAT_BGRA8 },
-        { rs_fourcc( 'M', 'J', 'P', 'G' ), RS2_FORMAT_MJPEG },
-        { rs_fourcc( 'C', 'N', 'F', '4' ), RS2_FORMAT_RAW8 },
-        { rs_fourcc( 'B', 'Y', 'R', '2' ), RS2_FORMAT_RAW16 },
-        { rs_fourcc( 'M', 'X', 'Y', 'Z' ), RS2_FORMAT_MOTION_XYZ32F },
+    static std::map< std::string, int > fcc_to_rs2{  // copy from ds5-device.cpp
+        { "YUY2", RS2_FORMAT_YUYV },
+        { "YUYV", RS2_FORMAT_YUYV },
+        { "UYVY", RS2_FORMAT_UYVY },
+        { "mono8", RS2_FORMAT_Y8 },  // Used by IR streams; ROS2-compatible
+        { "Y8I",  RS2_FORMAT_Y8I },
+        { "W10",  RS2_FORMAT_W10 },
+        { "Y16",  RS2_FORMAT_Y16 },
+        { "Y12I", RS2_FORMAT_Y12I },
+        { "Y16I", RS2_FORMAT_Y16I },
+        { "16UC1", RS2_FORMAT_Z16 },  // Used by depth streams; ROS2-compatible
```

Comment:
So we have some formats that are supported by ROS and some that don't?
What all GB are capitalized except rgb8?
Its the only one suppoerted by ROS?

Reply:
The formats are detailed here:
http://docs.ros.org/en/jade/api/sensor_msgs/html/image__encodings_8h_source.html
AFAIK, they're case-sensitive, and we stick to the same case.
These are our internal format that we use.
These particular functions are used to translate between rs2 and realdds, so are not very important. These 3 specific ones are the ones we actually use for depth, color, ir, and so have been tested with ROS. The others haven't been tested.

Reply:
OK  got it thanks